spec/index.adoc

```diff
@@ -619,29 +627,97 @@ See <<state-tree>> for details on the state tree.
 
 If the call resulted in a reject, the response is a CBOR map with the following fields:
 
-* `status` (`text`): `rejected`
+* `status` (`text`): `"rejected"`
 * `reject_code` (`nat`): The reject code (see <<reject-codes>>).
 * `reject_message` (`text`): a textual diagnostic message.
-* `error_code` (text): an optional implementation-specific textual error code (see <<error-codes>>).
+* `error_code` (`text`): an optional implementation-specific textual error code (see <<error-codes>>).
+
+[#http-signed-query]
+=== Request: Signed query call
+
+NOTE: The signed query call API is considered EXPERIMENTAL. Canister developers must be aware that the API may evolve in a non-backward-compatible way.
+
+A *signed* query call is a query call amended with signatures produced by IC nodes evaluating the query call to certify the query response.
+
+In order to make a signed query call to a canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/signed_query`.
+The request body has the same structure as the request body for <<http-query>> with the `content` map having the following *additional* fields:
+
+* (optional) `return_signatures` (`text`): Determines whether and how many signatures are returned in the response; possible values are `"none"`, `"one"`.
+* (optional) `return_certificate` (`text`): Determines whether a certificate (as specified in <<certification>>) is returned in the response; possible values are `"no"`, `"yes"`.
+
+The response is a CBOR (see <<cbor>>) with the same fields as for <<http-query>> response and the following *additional* fields:
+
+* (optional) `signatures` (`[+ node-signature]`): a non-empty list of node signatures for the returned query response.
+* (optional) `certificate` (`blob`): a certificate, as specified in <<certification>>.
+
+Unless `return_signatures` is null or set to `"none"` in the request, the response to a signed query call contains a non-empty list of signatures
+for the returned response produced by the individual IC nodes that computed the same returned response.
+Every such signature (whose type is denoted as `node-signature`) is a CBOR (see <<cbor>>) map with the following fields:
 
-Canister methods that do not change the canister state can be executed more efficiently. This method provides that ability, and returns the canister’s response directly within the HTTP response.
+* `timestamp` (`nat`): the timestamp of the signature.
+* `signature` (`blob`): the actual signature.
+* `identity` (`principal`): the principal of the node producing the signature.
+
+Unless `return_certificate` is null or set to `"no"` in the request, the response to a signed query call contains a certificate containing
+all paths in the state tree with prefixes
+
+* `/time`. Always contained in the certificate.
+* `/subnet/<subnet_id>/canister_ranges`. Contained in the certificate if the certificate does not have a subnet delegation, as specified in <<certification-delegation>>.
```

Comment:
@mraszyk Does this field means we don't include NNS self delegation in certificate but verify the canister ranges directly from the tree?

Reply:
Indeed. See here: https://github.com/dfinity/interface-spec/pull/163#discussion_r1232164262